lib/segment/src/common/mod.rs

```diff
@@ -45,7 +45,14 @@ fn check_query_vector(
     vector_config: &VectorDataConfig,
 ) -> OperationResult<()> {
     match query_vector {
-        QueryVector::Nearest(vector) => check_vector_against_config(vector, vector_config)?,
+        QueryVector::Nearest(vector) => {
```

Comment:
```suggestion
        QueryVector::Nearest(vector) => {
            check_vector_against_config(vector.to_vec_ref().into(), vector_config)?
        }
```

Reply:
I propose to use temporary`into()` conversion here. Instead of todo comment we will use compilation errors as marks what we have forgotten